app.js

```diff
@@ -2,24 +2,121 @@ const express = require('express');
 const path = require('path');
 const register = require('./api/registerRouter');
 const app = express();
+const mongoose = require('mongoose');
+const AuthRoute = require('./routes/auth')
+const session = require('express-session');
+const User = require('./models/user')
+const bcrypt = require('Bcryptjs')
+const jwt = require('jsonwebtoken')
+const MongoDBStore = require('connect-mongodb-session')(session);
+mongoose.Promise = global.Promise
 
-//body parser Midware
-app.use(express.json());
-app.use(express.urlencoded({ extended: false }));
+const mongoDBUri = 'mongodb+srv://PosterApp:A1s2f4g5%2A@poster-data.noemv.mongodb.net/Data'
 
-app.use('/static', express.static(path.join(__dirname, 'public')))
+mongoose.connect(mongoDBUri,
+{   useNewUrlParser:true,
+    useUnifiedTopology:true});
 
-app.get('/', (req, res) => {
+    mongoose.connection.once('open',function(){
+        console.log('connection is success!! ');
+    }).on('error', function(error){
+        console.log('***connection not available***',error);
+    });
+
+const store = new MongoDBStore({
+    uri: mongoDBUri,
+    collection: 'database'
+});
+app.use(express.urlencoded({extended: true}));
+
+app.use(session({
+    secret: 'slob that bob top',
+    resave: false,
+    saveUninitialized: false,
+    store: store,
+}))
+
+const isAuth = (req, res, next) => {
+    if (req.session.isAuth) {
+        next();
+    } else {
+        res.redirect('/')
+    }
+}
+
+app.get('/homepage',isAuth , (req, res) => {
     res.sendFile(path.resolve(__dirname,'public/homepage/homepage.html'))
 })
 
-app.get('/login', (req, res) => {
+app.get('/', (req, res) => {
     res.sendFile(path.resolve(__dirname,'public/login/login.html'))
 })
+
+app.post('/', async (req, res) => {
+    const { email, password} = req.body;
+
+    const user = await User.findOne({email});
+
+    if (!user) {
+        return  res.redirect('/');
+    }
+
+    const isMatch = await bcrypt.compare(password, user.password)
+
+    if (!isMatch) {
+        return  res.redirect('/');
+    }
+
+    req.session.isAuth = true
+    res.redirect('/homepage');
+})
```

Comment:
I am assuming that this is a login api, it has to be in the api routes file. not in app.js.

Also since this is an api, res.redirect is not gonna work, because the script is being redirected instead of the browser. You will need to handle it differently, by sending an object {success: true} or something and then in frontend check if it succeeded then redirect to "/" (homepage)

Reply:
fixed at publicJS